File: app/src/main/java/com/example/conversations/reviews/SubmitReviewActivity.java
package com.example.conversations.reviews;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import com.bazaarvoice.bvandroidsdk.Action;
import com.bazaarvoice.bvandroidsdk.AuthenticationProvider;
import com.bazaarvoice.bvandroidsdk.BVConversationsClient;
import com.bazaarvoice.bvandroidsdk.BVHostedAuthenticationProvider;
import com.bazaarvoice.bvandroidsdk.ConversationsSubmissionCallback;
import com.bazaarvoice.bvandroidsdk.ConversationsSubmissionException;
import com.bazaarvoice.bvandroidsdk.ReviewSubmissionRequest;
import com.bazaarvoice.bvandroidsdk.ReviewSubmissionResponse;
import com.example.conversations.App;
import com.example.conversations.Constants;
import com.example.conversations.R;

import static com.example.conversations.Util.submissionExceptionToString;

public class SubmitReviewActivity extends AppCompatActivity {
  private TextView submitResult;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_with_text);
    submitResult = findViewById(R.id.displayText);
    final BVConversationsClient bvClient = App.get(this).getBvClient();

    final AuthenticationProvider authProvider = new BVHostedAuthenticationProvider(Constants.UAS); // if BV Hosted Authentication
    // OR // = new SiteAuthenticationProvider(Constants.UAS); // if Site Authentication

    // Send Request
    final ReviewSubmissionRequest request = new ReviewSubmissionRequest
        .Builder(Action.Preview, Constants.PRODUCT_ID)
        .authenticationProvider(authProvider)
        .rating(5)
        .title(Constants.REVIEW_TITLE)
        .reviewText(Constants.REVIEW_TEXT)
        .userNickname(Constants.USER_NICKNAME)
        .build();
    bvClient.prepareCall(request).loadAsync(viewFormCb);
  }

  private final ConversationsSubmissionCallback<ReviewSubmissionResponse> viewFormCb = new ConversationsSubmissionCallback<ReviewSubmissionResponse>() {
    @Override
    public void onSuccess(@NonNull ReviewSubmissionResponse response) {
      submitResult.setText("Success!\n\nNow submit with Action.Submit to actually submit");
    }

    @Override
    public void onFailure(@NonNull ConversationsSubmissionException exception) {
      submitResult.setText("Error!\n\n" + submissionExceptionToString(exception));
    }
  };

  public static void transitionTo(Context fromContext) {
    fromContext.startActivity(new Intent(fromContext, SubmitReviewActivity.class));
  }
}

File: app/src/main/java/com/example/conversations/reviews/SubmitReviewActivity.kt
package com.example.conversations.reviews

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.bazaarvoice.bvandroidsdk.*
import com.example.conversations.App
import com.example.conversations.Constants
import com.example.conversations.R
import com.example.conversations.Util.submissionExceptionToString

class SubmitReviewActivity : AppCompatActivity() {
    private var submitResult: TextView? = null

    private val viewFormCb = object : ConversationsSubmissionCallback<ReviewSubmissionResponse> {
        override fun onSuccess(response: ReviewSubmissionResponse) {
            submitResult?.text = "Success!\n\nNow submit with Action.Submit to actually submit"
        }

        override fun onFailure(exception: ConversationsSubmissionException) {
            submitResult?.text = "Error!\n\n" + submissionExceptionToString(exception)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_with_text)
        submitResult = findViewById(R.id.displayText)
        val bvClient = App.get(this).bvClient

        val authProvider = BVHostedAuthenticationProvider(Constants.UAS) // if BV Hosted Authentication
        // OR // = new SiteAuthenticationProvider(Constants.UAS); // if Site Authentication

        // Send Request
        val request = ReviewSubmissionRequest.Builder(Action.Preview, Constants.PRODUCT_ID)
                .authenticationProvider(authProvider)
                .rating(5)
                .title(Constants.REVIEW_TITLE)
                .reviewText(Constants.REVIEW_TEXT)
                .userNickname(Constants.USER_NICKNAME)
                .build()
        bvClient!!.prepareCall(request).loadAsync(viewFormCb)
    }

    companion object {

        fun transitionTo(fromContext: Context) {
            fromContext.startActivity(Intent(fromContext, SubmitReviewActivity::class.java))
        }
    }
}
